File: 0026_removeDuplicatesFromSortedArray/main.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        int l = 0, r = 0;
        while(r < nums.size()) {
            if(nums[l] == nums[r]) {
                ++r;
            } else {
                ++l;
                nums[l] = nums[r];
            }
        }
        return l + 1;
    }

    int removeDuplicates1(std::vector<int>& nums) {
        int count = 0;
        auto it = std::unique(nums.begin(), nums.end());
        nums.erase(it, nums.end());
        return nums.size();
    }

    int removeDuplicates2(std::vector<int>& nums) {
        if(nums.size() == 0)
            return 0;
        if(nums.size() == 1)
            return 1;
        for(int i = 1; i < nums.size(); ++i) {
            if(nums[i] == nums[i-1]) {
                nums.erase(nums.begin()+i);
                --i;
            }
        }
        return nums.size();
    }
};

File: 0026_removeDuplicatesFromSortedArray/main.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        r = 1
        while r < len(nums):
            if nums[l] == nums[r]:
                r += 1
                continue
            else:
                l += 1
                nums[l] = nums[r]
        return l + 1
    
    def removeDuplicates1(self, nums: List[int]) -> int:
        nums[:] = sorted(set(nums))
        return len(nums)
    
    def removeDuplicates2(self, nums: List[int]) -> int:
        i = 1
        while i < len(nums):
            if nums[i] == nums[i-1]:
                nums.pop(i)
            else:
                i += 1
        return len(nums)

test = [1,1,2]

Solution().removeDuplicates(test)